### PR TITLE
feat(sync): add browsing and viewing commands to sync plugin

### LINKED ISSUE #48: feat: Add specs browsing and viewing commands to sync plugin
## Summary
Add new commands to the sync plugin for easily browsing and viewing specs with their sync status and frontmatter metadata.

## Motivation
Currently, users need to manually navigate the file system to view specs. Adding dedicated commands for listing, viewing, and browsing specs would improve the developer experience by providing:
- Quick overview of all specs and their sync status
- Easy access to spec details and frontmatter
- Interactive browsing capabilities

### Extend the sync plugin with new commands:

1. **`specify sync list`** - List all specs with status
   - Display specs in a tree structure
   - Show sync status indicators (synced/draft/conflict)
   - Display key frontmatter fields (issue number, last sync)
   - Color-coded output based on status

2. **`specify sync view [spec-path]`** - View spec details
   - Display full spec content
   - Show all frontmatter metadata
   - List files in spec directory
   - Display sync history if available

3. **`specify sync browse`** - Interactive browser
   - Navigate specs with arrow keys
   - Preview spec content inline
   - Quick actions (sync, edit, open in GitHub)
   - Search and filter capabilities

### 1. Command Structure
```
plugins/sync/src/commands/
├── list.command.ts     # List all specs
├── view.command.ts     # View single spec
└── browse.command.ts   # Interactive browser
```

### 2. UI Components
```
plugins/sync/src/ui/
├── SpecTreeView.ts     # Tree visualization
├── SpecDetailsView.ts  # Detailed display
└── SpecBrowser.ts      # Interactive TUI
```

### 3. Integration Points
- Reuse existing `SpecScanner` class for reading specs
- Leverage current frontmatter parsing logic
- Integrate with sync status from adapters
- Extend `createSyncCommand()` in index.ts

## Benefits
- **Improved UX**: Easier navigation and discovery of specs
- **Better visibility**: Clear view of sync status across all specs
- **Faster workflow**: Quick access to spec information without file browsing
- **Code reuse**: Leverages existing sync plugin infrastructure

## Testing Requirements
- Unit tests for new commands
- UI component rendering tests
- Integration tests with existing sync functionality
- Edge cases (empty specs, missing frontmatter, malformed files)

## Future Enhancements
- Export specs list to various formats (JSON, CSV)
- Bulk operations on multiple specs
- Advanced filtering and search
- Integration with other plugins

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

### PR DESCRIPTION
**Pull request by amondnet**
## Summary by CodeRabbit

- New Features
  - Interactive Spec Browser with navigation, previews, and actions (sync, edit, open in GitHub).
  - New CLI commands: sync browse, sync list (with filtering), sync view (detailed spec view).
  - Enhanced sync status (adds “local”) and last-checked timestamps; improved table/detail presentations.

- Documentation
  - Expanded usage docs for unified specify sync CLI with examples.
  - Added extensive templates and guides for specs, planning, tasks, agent contexts, and constitution/change checklist.

- Tests
  - Added unit tests for list command, SpecTable, and SpecDetails.

- Chores
  - New helper scripts and updated build/dependency config to support tooling.

### CODE REVIEW

File: plugins/sync/src/ui/SpecBrowser.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        return 'exit'
+      default:
+        return 'back'
+    }
+  }
+
+  private async syncSpec(spec: SpecDocument): Promise<void> {
+    console.log(chalk.blue(`🔄 Syncing ${spec.name}...`))
+    // This would call the actual sync functionality
+    // For now, just show a message
```
Comment (amondnet) [inline plugins/sync/src/ui/SpecBrowser.ts]: @claude add sync feature

**Comment by amondnet** [comment]
@claude fix lint, typecheck, test
